packages/material-ui/src/Popper/Popper.js

```diff
@@ -121,25 +121,43 @@ const Popper = React.forwardRef(function Popper(props, ref) {
       }
     }
 
-    const popper = new PopperJs(getAnchorEl(anchorEl), tooltipRef.current, {
+    let popperModifiers = [
+      {
+        name: 'preventOverflow',
+        options: {
```

Comment:
Before altBoundary also affected flip but the behaviour is now customisable in both places.